src/sota_tools/garage_deploy.cc

```diff
@@ -112,20 +112,24 @@ int main(int argc, char **argv) {
       return EXIT_FAILURE;
     }
 
-    if (mode == RunMode::kDefault) {
-      if (push_credentials.CanSignOffline()) {
-        bool ok = OfflineSignRepo(ServerCredentials(push_credentials.GetPathOnDisk()), name, commit, hardwareids);
-        if (ok) {
-          if (CheckRefValid(fetch_server, ostree_commit, mode, max_curl_requests) != EXIT_SUCCESS) {
-            LOG_FATAL << "Check if the ref is present on the server or in targets.json failed";
-            return EXIT_FAILURE;
-          }
-        } else {
-          return EXIT_FAILURE;
-        }
+    if (mode == RunMode::kDefault || mode == RunMode::kPushTree) {
+      if (!push_credentials.CanSignOffline()) {
+        LOG_FATAL << "Online signing with garage-deploy is not supported.";
+        return EXIT_FAILURE;
+      }
+      if (!OfflineSignRepo(ServerCredentials(push_credentials.GetPathOnDisk()), name, commit, hardwareids)) {
+        return EXIT_FAILURE;
+      }
+
+      TreehubServer push_server;
+      if (authenticate(cacerts, push_credentials, push_server) != EXIT_SUCCESS) {
```

Comment:
Shouldn't  push_server authenticate be done before 'UploadToTreehub'?

Reply:
Yeah, I've accidentally made it authenticate twice now. I'll try to refactor this.

Reply:
Fixed.